### PR TITLE
Update notes entry for new dynamic runner loading

### PR DESCRIPTION
**Pull request by Kami**
Upgrade notes entry for change in https://github.com/StackStorm/st2/pull/4217.

### CODE REVIEW

File: docs/source/upgrade_notes.rst [hunk 1]
```diff
@@ -3,6 +3,39 @@
...
+
+  In this release we updated our runner loading mechanism which makes ``/opt/stackstorm/runners``
+  directory obsolete.
+
+  All the runners are now installed as Python packages into |st2| virtual environment
+  (``/opt/stackstorm/st2``) during package build process and dynamically loaded when requested.
+
+  This provides for more flexible installation and loading of runner modules. To install a custom
+  runner, user now just needs to install Python package which contains runner module into |st2|
+  virtual environment and run ``sudo st2ctl reload --register-runners`` command.
```
Comment (armab) [inline docs/source/upgrade_notes.rst]: Just to be sure I'm understanding it right, default runners that we package in core don't need additional `--register-runners` and would work by default on app init?

Comment (Kami) [reply docs/source/upgrade_notes.rst]: This command still needs to be run so the information from disk is stored in the database (same as for any other resources).

In the future though, we could make that step implicit and happen automatically on service start up (like it used to be the case).

Comment (armab) [reply docs/source/upgrade_notes.rst]: In this case, it sounds like a regression to me :-1: 

While ideally to go towards pre-init all the content that comes with st2 core on app startup, making the st2 self-sufficient and minimally working after the first installation (and that really affects all deployments), we sadly go the opposite side offloading repetitive instructions on user's side adding more complexity.

Comment (Kami) [reply docs/source/upgrade_notes.rst]: It's not a regression - it's exactly the same end user behavior, just the internals change.

While I would prefer if we could get rid of it, we need to approach it in incremental steps.

Comment (Kami) [reply docs/source/upgrade_notes.rst]: Having said that, I will see what we can do.

That st2 PR is already growing a bit big and I would like to scope things down.

**Pull request merged**